### ISSUE TITLE
players in multiplayer slide instead of walk

### ISSUE THREAD

**Issue by Bem20** [Yesssssman/epicfightmod#1164]
I see how other players slide instead of walking, it only happens with other players in multiplayer, in my camera my character makes the animation walk correctly 

https://user-images.githubusercontent.com/84411058/187282441-0f50adeb-0a20-4ba0-b4c4-23b65f49fe0a.mp4

Mod List:
		dynamiclightsreforged-1.18.2_v1.3.3.jar           
		YungsBetterDungeons-1.18.2-Forge-2.1.0.jar        
		YungsBetterWitchHuts-1.18.2-Forge-1.0.1.jar       
		majruszs-difficulty-1.18.2-1.4.4.jar              
		DoggyTalents-1.18.2-2.5.0.jar                     
		Entity_Collision_FPS_Fix-forge-1.18.2-1.0.0.jar   
		Incendium_v5.0.4.jar                              
		sophisticatedcore-1.18.2-0.5.1.98.jar             
		rubidium-0.5.3a.jar                               
		Controlling-forge-1.18.2-9.0+22.jar               
		citadel-1.11.3-1.18.2.jar                         
		alexsmobs-1.18.6.jar                              
		YungsApi-1.18.2-Forge-2.0.8.jar                   
		feature_nbt_deadlock_be_gone_forge-2.0.0+1.18.2.ja
		sophisticatedbackpacks-1.18.2-3.18.17.680.jar     
		guardvillagers-1.18.2.1.4.3.jar                   
		buildinggadgets-3.13.0-build.5+mc1.18.2.jar       
		bygonenether-1.2.1-1.18.2.jar                     
		balm-3.2.0+0.jar                                  
		carryon-1.18.2-1.17.0.8.jar                       
		JustEnoughResources-1.18.2-0.14.1.171.jar         
		cloth-config-6.2.62-forge.jar                     
		soundphysics-forge-1.18.2-1.0.6.jar               
		dragonmounts-1.18.2-1.1.4.jar                     
		PrettyPipes-1.12.6.jar                            
		structure_gel-1.18.2-2.4.6.jar                    
		flytre_lib-1.18.2-1.8.2-forge.jar                 
		UndeadExpansion_2.0.4_1.18.2_b.jar                
		FarmersDelight-1.18.2-1.1.2.jar                   
		Aquamirae 3.2.0 (Forge 1.18.2).jar                
		braincell-1.18.jar                                
		ironfurnaces-1.18.2-3.3.1.jar                    
		YungsBridges-1.18.2-Forge-2.1.0.jar               
		wthit-forge-4.13.3.jar                            
		curiouslanterns-1.18.2-1.1.0.jar                  
		dpanvil-1.18.2-3.4.1.jar                          
		curios-forge-1.18.2-5.0.7.1.jar                  
		Cataclysm-0.19.jar                                
		Patchouli-1.18.2-71.1.jar                         
		blockui-1.18.2-0.0.58-ALPHA.jar                   
		EasyPaxelLite1.18.2(FORGE)vs1.0.14.jar            
		YungsExtras-1.18.2-Forge-2.1.0.jar                
		elevatorid-1.18.2-1.8.4.jar                       
		YungsBetterStrongholds-1.18.2-Forge-2.1.1.jar     
		tombstone-7.4.11-1.18.2.jar                       
		EnigmaticLegacy-2.24.1.jar                        
		worldedit-mod-7.2.10.jar                          
		cfm-7.0.0-pre29-1.18.2.jar                        
		architectury-4.9.83-forge.jar                     
		curiouselytra-forge-1.18.1-5.0.1.0.jar            
		majrusz-library-1.18.2-2.7.3.jar                  
		inventoryessentials-forge-1.18.2-4.0.2.jar       
		krypton-0.1.10-SNAPSHOT.jar                       
		YungsBetterMineshafts-1.18.2-Forge-2.2.jar        
		PlayerRevive_FORGE_v2.0.9_mc1.18.2.jar            
		geckolib-forge-1.18-3.0.36.jar                    
		EnchantingInfuser-v3.3.2-1.18.2-Forge.jar         
		DripSounds-1.18-0.3.0.jar                         
		clienttweaks-forge-1.18.1-7.1.0.jar               
		trashslot-forge-1.18.1-11.0.2.jar                 
		minecraft-comes-alive-7.3.20+1.18.2-universal.jar 
		Barbarians-1.18.2-1.0.5.jar                       
		EasyMagic-v3.3.0-1.18.2-Forge.jar                 
		jei-1.18.2-9.7.1.255.jar                          
		JEIEnchantmentInfo-1.17.1-2.0.0.jar               
		the-conjurer-1.18.2-1.1.1.jar                     
		caelus-forge-1.18.1-3.0.0.2.jar                   
		Paxi-1.18.2-Forge-2.0.jar                         
		waystones-forge-1.18.2-10.1.0.jar                 
		EpicFight-18.3.3-beta.jar                        
		Clumps-forge-1.18.2-8.0.0+15.jar                  
		journeymap-1.18.2-5.8.5-forge.jar                 
		dual_riders-1.18.2-1.1.1-forge.jar                
		comforts-forge-1.18.2-5.0.0.4.jar                 
		artifacts-1.18.2-4.0.7.jar                        
		Serene Seasons-1.18.2-7.0.0.13.jar                
		configured-1.5.4-1.18.2.jar                       
		badpackets-forge-0.1.2.jar                        
		BadMobs-1.18.2-13.01.jar                          
		Obscuria's Tooltips 1.4.1 (1.18.2).jar            
		snowundertrees-1.18.2-1.2.1.jar                   
		lazydfu-1.0-1.18+.jar                             
		YungsBetterDesertTemples-1.18.2-Forge-1.3.1.jar   
		walljump-forge-1.18.1-1.3.7.jar                   
		blueprint-1.18.2-5.4.3.jar                        
		savage_and_ravage-1.18.2-4.0.0.jar                
		forge-1.18.2-40.1.73-universal.jar                
		curiouslights-forge-1.2.0+1.18.2.jar              
		idas_forge-1.4.1+1.18.2.jar                       
		selene-1.18.2-1.17.9.jar                          
		supplementaries-1.18.2-1.4.12.jar                 
		ironchest-1.18.2-13.1.9.jar                       
		DungeonsArise-1.18.2-2.1.50d-beta.jar             
		client-1.18.2-20220404.173914-srg.jar             
		TConstruct-1.18.2-3.5.1.31.jar                    
		sons-of-sins-1.18.2-1.0.9.jar                     
		illuminations-forge-1.18.2-0.0.4.jar              
		CreativeCore_FORGE_v2.6.15_mc1.18.2.jar           
		weaponmaster-multi-forge-1.18.1-3.0.3.jar         
		domum_ornamentum-1.18.2-1.0.50-ALPHA-universal.jar
		flywheel-forge-1.18.2-0.6.5.jar                   
		create-1.18.2-0.5.0.d.jar                         
		Mantle-1.18.2-1.9.27.jar                          
		brutalbosses-1.18.2-5.4.jar                       
		polymorph-forge-1.18.2-0.44.jar                   
		AutoRegLib-1.7-53.jar                             
		Quark-3.2-358.jar                                 
		DramaticDoors-Forge-1.18.2-2.1.3.jar              
		entityculling-forge-mc1.18-1.5.1.jar              
		structurize-1.18.2-1.0.424-ALPHA.jar              
		multi-piston-1.18.2-1.2.15-ALPHA.jar              
		minecolonies-1.18.2-1.0.966-RELEASE.jar           
		HunterIllager-1.18.2-4.0.1.jar                    
		dannys_expansion-1.2.6.jar                        
		creeperoverhaul-1.18.2-1.3.0-forge.jar            
		ferritecore-4.2.1-forge.jar                       
		PuzzlesLib-v3.3.5-1.18.2-Forge.jar                
		Obscuria's Essentials 3.2.0 (Forge 1.18.2).jar    
		expandability-6.0.0.jar                           
		CosmeticArmorReworked-1.18.2-v2.jar               
		domesticationinnovation-1.4.2-1.18.2.jar      

Thanks for your support.

**Comment by MetalKnight56** [comment]
Seems to be happening on lan servers without any mods 

https://user-images.githubusercontent.com/77132244/187794073-21776f23-bbb3-4b60-8fd8-b07ac9015bf7.mp4

**Comment by uoissvd** [comment]
Hi

This issue seems to do with changes made in _AbstractClientPlayerPatch.java_ since version 18.3.0.

In the latest version, 18.3.3, player movement is checked to determine the currentLivingMotion.

```java
else if (original.zza != 0.0F || original.xxa != 0.0F) {
	if (original.isShiftKeyDown())
		currentLivingMotion = LivingMotions.SNEAK;
	else if (original.isSprinting())
		currentLivingMotion = LivingMotions.RUN;
	else
		currentLivingMotion = LivingMotions.WALK;
	
	if (original.zza < 0)
		animator.baseLayer.animationPlayer.setReversed(true);
	else 
		animator.baseLayer.animationPlayer.setReversed(false);
	
} else {
```

It is using original.zza and original.xxa which seems to be only updating for LocalPlayer not AbstractClientPlayer. Hence you don't notice the issue with a single player.

Versions prior to 18.3.0 used original.animationSpeed to determine the currentLivingMotion.

I believe the change was made to call the setReversed() method depending on the movement direction.
I don't write mods so I don't know what alternative methods are available to check the same condition but I'm hoping this could help to resolve the issue.

**Comment by Yesssssman** [comment]
Fixed in 18.3.4
